src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/AsyncInductionEventService.kt

```diff
@@ -35,23 +37,56 @@ class AsyncInductionEventService(
 
   private fun buildInductionCreatedEvent(induction: Induction): TimelineEvent =
     with(induction) {
+      val noteContent = note?.content ?: ""
+      val conductedByPerson = conductedBy ?: ""
+      val conductedByRolePerson = conductedByRole ?: ""
+
       TimelineEvent.newTimelineEvent(
         sourceReference = reference.toString(),
         eventType = TimelineEventType.INDUCTION_CREATED,
         prisonId = createdAtPrison,
         actionedBy = induction.createdBy!!,
         timestamp = induction.createdAt!!,
+        contextualInfo = mapOf(
+          TimelineEventContext.COMPLETED_INDUCTION_ENTERED_ONLINE_AT to createdAt.toString(),
+          TimelineEventContext.COMPLETED_INDUCTION_ENTERED_ONLINE_BY to userService.getUserDetails(createdBy!!).name,
+          TimelineEventContext.COMPLETED_INDUCTION_CONDUCTED_IN_PERSON_DATE to completedDate.toString(),
+          TimelineEventContext.COMPLETED_INDUCTION_NOTES to noteContent,
+          *conductedBy
+            ?.let {
+              arrayOf(
+                TimelineEventContext.COMPLETED_INDUCTION_CONDUCTED_IN_PERSON_BY to conductedByPerson,
+                TimelineEventContext.COMPLETED_INDUCTION_CONDUCTED_IN_PERSON_BY_ROLE to conductedByRolePerson,
+              )
+            } ?: arrayOf(),
+        ),
       )
     }
 
   private fun buildInductionUpdatedEvent(induction: Induction): TimelineEvent =
     with(induction) {
+      val noteContent = note?.content ?: ""
+      val conductedByPerson = conductedBy ?: ""
+      val conductedByRolePerson = conductedByRole ?: ""
       TimelineEvent.newTimelineEvent(
         sourceReference = reference.toString(),
         eventType = TimelineEventType.INDUCTION_UPDATED,
         prisonId = lastUpdatedAtPrison,
         actionedBy = induction.lastUpdatedBy!!,
         timestamp = induction.lastUpdatedAt!!,
+        contextualInfo = mapOf(
```

Comment:
I dont think we want to do this for the update event 🤔  The UI timeline event design for updating the induction doesnt need to include any of these extra fields (there are no requirements to change how the timeline event is displayed for the Induction updated event)

Reply:
ah ok